backend/src/v2/driver/driver.go

```diff
@@ -1694,10 +1695,17 @@ func resolveUpstreamParameters(cfg resolveUpstreamOutputsConfig) (*structpb.Valu
 		return nil, cfg.err(fmt.Errorf("output parameter key is empty"))
 	}
 
+	isParallelForDAG := cfg.dag.Execution.GetExecution().GetCustomProperties()["iteration_index"] != nil
```

Comment:
It might be helpful to include a comment here explaining what this block is doing. In particular, it might be helpful to provide some context on what `iteration_index`, how parallelFor's use it, and why we need to iterate using it.